### PR TITLE
feat: allow snake_case (alongside camelCase) for config dict

### PR DESCRIPTION
**Pull request by michdr**
# Pull Request

## Changes

- Allow the conventional pythonic use of `snake_case` within any config dict.
- Return `config` dict with the same expected `snake_case` conventions.

## Why this change?

Easier and more intuitive usage.

## Testing

How was the change tested?

## Additional Notes

Any additional information or context you want to provide?

### CODE REVIEW

**Comment by shuangwu5** [comment]
Would it be simpler to handle this with Pydantic model?
`config = SomeConfig.model_validate(input_dict)` accepts both snake_case and camelCase.
`config.model_dump_json()` can return snake_case dict if `by_alias=False` or camelCase if `by_alias=True`.

**Comment by shuangwu5** [comment]
Hmm, they're actually defined in the public API (like [CreateGeneratorRequest](https://github.com/mostly-ai/mostly-app-v2/blob/dev/public-api/public-api.yaml#L1638-L1695), [CreateConnectorRequest](https://github.com/mostly-ai/mostly-app-v2/blob/dev/public-api/public-api.yaml#L1598-L1620)). But indeed I don't know how to make code generator populated those 😕

File: mostlyai/base.py [hunk 1]
```diff
@@ -73,6 +78,8 @@ def request(
         response_type: type = dict,
         raw_response: bool = False,
         is_api_call: bool = True,
+        do_json_camel_case=True,
+        do_response_dict_snake_case=True,
```
Comment (shuangwu5) [inline mostlyai/base.py:82]: type hint is missing

File: mostlyai/naming_conventions.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def snake_to_camel(snake_str: str) -> str:
+    components = snake_str.split("_")
+    return components[0] + "".join(x.title() for x in components[1:])
+
+
+def camel_to_snake(camel_str: str) -> str:
+    return re.sub(r"(?<!^)(?=[A-Z])", "_", camel_str).lower()
+
+
+def convert_case(input_data: dict, conv_func) -> dict:
```
Comment (shuangwu5) [inline mostlyai/naming_conventions.py:13]: same here (for `conv_func`)

**Review by shuangwu5 — APPROVED** [review]

File: mostlyai/base.py (hunk 1 above)
Comment (michdr) [reply mostlyai/base.py]: Thanks, done!

**Pull request merged**